modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesContextStore.java

```diff
@@ -113,6 +114,16 @@ void storeDurableSubscription(AndesSubscription subscription)
     void storeNodeDetails(String nodeID, String data) throws AndesException;
 
     /**
+     * Store cluster events such as QUEUE_ADDED, BINDING_ADDED, SUB_REMOVED, etc in the database.
+     *
+     * @param eventType    A string defining the type of the even. e.g., QUEUE_ADDED
+     * @param eventDetails Event details encoded as a String
+     * @param timestamp    TODO remove this
+     * @throws AndesException
+     */
+    void storeClusterEvent(String eventType, String eventDetails, long timestamp) throws AndesException;
```

Comment:
I guess it's not about converting the enum to a string when storing (there is no conversion as well if done properly with an enum). Most cases it is good to have the relevant type than a generic string. Less prone to errors on passing invalid values.